app/src/main/cpp/Device.h

```diff
@@ -90,6 +91,27 @@ struct EyeRect {
 
 };
 
+inline device::DeviceType GetDeviceTypeFromSystem(bool is6DoF) {
+  char model[128];
+  int length = PopulateDeviceModelString(model);
+
+  if (!strncmp(model, "Quest 2", length)) {
```

Comment:
Nit: If we are going to return on every branch, perhaps it'd be clearer to avoid the 'else if' clauses. 

Sorry if my question is too obvious, but I lack some context here, but aren't we doing too much work in case of the model being "Quest Pro" ? I mean that the code is going to compare the same string with different values; couldn't we just have a map, in a class field ideally, that we can use to improve this pattern matching ?

Reply:
Yeah I agree with both comments. Maybe we can have a map with hashes for the names